Core: Switch tests to Junit5 in catalog,encryption,inmemory,io,view

skytin1004 opened the pull request:
This pull request is related to the issue "Move JUnit4 tests to JUnit5 https://github.com/apache/iceberg/issues/7160".
I switched the files in [catalog,encryption,inmemory,io,view] pakages to JUnit5.

skytin1004 commented on core/src/test/java/org/apache/iceberg/io/TestByteBufferInputStreams.java:
```diff
@@ -55,18 +56,22 @@ public void testReadAll() throws Exception {
     for (int i = 0; i < bytes.length; i += 1) {
-      Assert.assertEquals("Byte i should be i", i, bytes[i]);
-      Assert.assertEquals("Should advance position", 35, stream.getPos());
+      Assertions.assertThat(bytes[i]).as("Byte i should be i").isEqualTo((byte) i);
```
This casting[(byte) i] is required. This is because in JUnit 4, the `assertEquals` method performs automatic type conversion when comparing values of different types. But in AssertJ, the `assertThat` method does not perform automatic type conversion.

nastra commented on core/src/test/java/org/apache/iceberg/inmemory/TestInMemoryInputFile.java:
```diff
@@ -22,16 +22,17 @@
     InputStream inputStream = inputFile.newStream();
-    Assert.assertEquals('a', inputStream.read());
+    Assertions.assertThat(inputStream.read())
+        .as("Checking read value from inputStream")
```
do we really need this description?

skytin1004 replied:
I think it's an unnecessary description.
I deleted unnecessary description.

nastra commented on core/src/test/java/org/apache/iceberg/io/TestByteBufferInputStreams.java:
```diff
@@ -506,7 +562,8 @@ public void testMarkDoubleReset() throws Exception {
 
     stream.mark(5);
-    Assert.assertEquals("Should read 5 bytes", 5, stream.read(new byte[5]));
+
```
unnecessary newline

skytin1004 replied:
I deleted the unnecessary newline.

nastra commented on core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java:
```diff
@@ -168,70 +166,71 @@ protected boolean supportsNamesWithSlashes() {
     Map<String, String> createProps = ImmutableMap.of("prop", "val");
     catalog.createNamespace(NS, createProps);
-    Assert.assertTrue("Namespace should exist", catalog.namespaceExists(NS));
+
```
unnecessary newline

skytin1004 replied:
I deleted the unnecessary newline.

skytin1004 commented on core/src/test/java/org/apache/iceberg/io/TestSingleBufferInputStream.java:
```diff
@@ -40,8 +41,9 @@ protected ByteBufferInputStream newStream() {
-    Assert.assertEquals("Position should not change", 0, DATA.position());
-    Assert.assertEquals("Limit should not change", DATA.array().length, DATA.limit());
+    assertThat(DATA.position()).as("Position should not change").isEqualTo(0);
+
```
I deleted the unnecessary newline.

skytin1004 commented on core/src/test/java/org/apache/iceberg/io/TestMultiBufferInputStream.java:
```diff
@@ -62,62 +63,66 @@ public void testSliceData() throws Exception {
-    Assert.assertEquals(0, five.position());
-    Assert.assertEquals(3, five.limit());
-    Assert.assertEquals(3, five.capacity());
+
```
I deleted the unnecessary newline.

nastra commented on core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java:
```diff
@@ -548,20 +556,20 @@ public void testTableNameWithDot() {
 
     catalog.buildTable(ident, SCHEMA).create();
-    Assert.assertTrue("Table should exist", catalog.tableExists(ident));
+
```
nit: unnecessary newline

skytin1004 commented:
I should have done gitpull in local after merging with Apache master branch, but I skipped the gil pull. 
After that I deleted unnecessary newlines and force push. 
So I Merge the  'apache:master' branch into test2.

nastra commented on core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java:
```diff
@@ -614,20 +621,23 @@ public void testCompleteCreateTable() {
+        .isEqualTo(TABLE_WRITE_ORDER);
+    Assertions.assertThat(table.properties().entrySet())
+        .as("Table properties should be a superset of the requested properties")
+        .containsAll(properties.entrySet());
```
@skytin1004 can you make that change please so that we can get this in?

nastra replied:
@skytin1004 this still uses `containsAll` rather than `containsExactlyInAnyOrder`. We want to make sure that `table.properties()` doesn't contain any more elements, hence the `containsExactlyInAnyOrder` check

skytin1004 commented:
Thanks for your review @nastra! I have reviewed all the code in this PR and changed the code to use the `.containsAll` method. 
In some parts of the code, I changed the code to use `the containsExactlyInAnyOrderElementsOf` method.

nastra commented on core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java:
```diff
@@ -651,24 +661,27 @@ public void testLoadTable() {
+        .isEqualTo(TABLE_WRITE_ORDER);
+    Assertions.assertThat(table.properties().entrySet())
+        .as("Table properties should be a superset of the requested properties")
+        .containsAll(properties.entrySet());
```
this still needs to be changed

nastra commented on core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java:
```diff
@@ -1478,33 +1527,39 @@ public void testCompleteCreateTransaction() {
+        .isEqualTo(TABLE_WRITE_ORDER);
+    Assertions.assertThat(table.properties().entrySet())
+        .as("Table properties should be a superset of the requested properties")
+        .containsAll(properties.entrySet());
```
`containsExactlyInAnyOrder` should be used here

nastra commented on core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java:
```diff
@@ -2489,30 +2617,28 @@ public void tableCreationWithoutNamespace() {
   private static void assertEmpty(String context, Catalog catalog, Namespace ns) {
     try {
-      Assert.assertEquals(context, 0, catalog.listTables(ns).size());
+      Assertions.assertThat(catalog.listTables(ns).size()).as("context").isEqualTo(0);
```
```suggestion
      Assertions.assertThat(catalog.listTables(ns)).as(context).isEmpty();
```
